### PR TITLE
Docs/describe debug levels

### PR DESCRIPTION
**Pull request by EugeniyBurmistrov**

### CODE REVIEW

File: docs/debugging.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
+---
+id: debugging
+title: Debugging
+---
+
+Resolve uses the [resolve-debug-levels](https://www.npmjs.com/package/resolve-debug-levels) package to log debug information. This package extends the [debug](https://www.npmjs.com/package/debug) library's functionality with logging levels. The package provides the following logging levels:
```
Comment (dirk-pieterse) [inline docs/debugging.md]:
```suggestion
Resolve uses the [resolve-debug-levels](https://www.npmjs.com/package/resolve-debug-levels) package to log debug information. This package extends the [debug](https://www.npmjs.com/package/debug) library's functionality with the following logging levels:
```

File: docs/debugging.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+| Level   | Description                                            |
+| ------- | ------------------------------------------------------ |
+| log     | Messages that should always be displayed               |
+| error   | Errors that can hinder normal program execution        |
```
Comment (dirk-pieterse) [inline docs/debugging.md]:
```suggestion
| error   | Errors that can prevent normal program execution        |
```

File: docs/debugging.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+| warn    | Potential problems in the application's implementation |
+| debug   | Information displayed for debugging purposes           |
+| info    | Messages that describe the current operation           |
+| verbose | Extended descriptions to the previous message          |
```
Comment (dirk-pieterse) [inline docs/debugging.md]: "to the previous message" Why 'to'?

File: docs/debugging.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+## Debug ReSolve
+
+The reSolve framework uses the `resolve` prefix for all its debugging namespaces. To enable the framework's debug output, assign `resolve*` to the `DEBUG` environment variable as shown below:
+
+```
+DEBUG=resolve:* DEBUG_LEVEL=error yarn dev
+```
+
+The reSolve framework displays messages of the **log**, **error** and **warn** levels by default.
```
Comment (dirk-pieterse) [inline docs/debugging.md]:
```suggestion
The reSolve framework displays **log**, **error**, and **warn** level messages.
```

File: docs/debugging.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+## Debug a ReSolve Application
+
+You can use [resolve-debug-levels](https://www.npmjs.com/package/resolve-debug-levels) to debug your reSolve application. To do this, add the `resolve-debug-levels` package to your application's dependencies:
```
Comment (dirk-pieterse) [inline docs/debugging.md]:
```suggestion
You can add the `resolve-debug-levels` package to your application's dependencies to use [resolve-debug-levels](https://www.npmjs.com/package/resolve-debug-levels) to debug your reSolve application.
```

File: docs/debugging.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+```
+yarn add resolve-debug-levels
+```
+
+To create a logger, pass your module's debugging namespace to the function exposed by the `resolve-debug-levels` module:
```
Comment (dirk-pieterse) [inline docs/debugging.md]:
```suggestion
To create a logger, pass your module's debugging namespace to the function the `resolve-debug-levels` module exposes:
```

File: docs/debugging.md [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+
+```js
+import debugLevels from 'resolve-debug-levels'
+const log = debugLevels('myapp:api-handlers')
+...
+```
+
+The logger object exposes methods that correspond to the available debug levels:
```
Comment (dirk-pieterse) [inline docs/debugging.md]: "to the available debug" Why 'the available'?

Comment (EugeniyBurmistrov) [reply docs/debugging.md]: changed to "to the debug levels"

File: docs/debugging.md (hunk 3 above)
Comment (EugeniyBurmistrov) [reply docs/debugging.md]: Changed to 'Extended descriptions for the previous message'

Comment (dirk-pieterse) [reply docs/debugging.md]: Ok

File: docs/debugging.md (hunk 7 above)
Comment (dirk-pieterse) [reply docs/debugging.md]: Ok

**Pull request merged**